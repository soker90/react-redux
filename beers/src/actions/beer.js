import beerAction from '../constants/actions/beer'

export default {
  fetchBeer: () => ({type: beerAction.FETCH_BEER}),
  fetchBeerSuccess: beers => ({type: beerAction.FETCH_BEER_SUCCESS, beers}),
  addBeer: (name, graduation, price) => ({type: beerAction.ADD_BEER, name, graduation, price}),
  addBeerSuccess: beer => ({type: beerAction.ADD_BEER_SUCCESS, beer}),
  removeBeer: beerId => ({type: beerAction.REMOVE_BEER, beerId}),
  removeBeerSuccess: beerId => ({type: beerAction.REMOVE_BEER_SUCCESS, beerId}),
  beerError: error => ({type: beerAction.BEER_ERROR, error})
}
