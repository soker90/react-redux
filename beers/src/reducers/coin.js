import coinAction from '../constants/actions/coin'

export function initialState () {
  return {
    coinList: [],
    coin: {},
    loading: false,
    error: null
  }
}

function fetchCoin(state) {
  return {
    ...state,
    coinList: [],
    loading: true,
    error: null
  }
}

function fetchCoinSuccess(state, {coins}) {
  return {
    ...state,
    coinList: coins.coins,
    loading: false,
    error: null
  }
}

function addCoin(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function addCoinSuccess(state, {coin}) {
  return {
    ...state,
    coinList: [coin, ...state.coinList],
    loading: false,
    error: null
  }
}

function removeCoin(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function removeCoinSuccess(state, {coinId}) {
  return {
    ...state,
    coinList: state.coinList.filter((coin) => coin.id !== coinId.coinId),
    loading: false,
    error: null
  }
}

function coinError(state, {error}) {
  return {
    ...state,
    loading: false,
    error: error
  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case coinAction.FETCH_COIN:
      return fetchCoin(state, action)
    case coinAction.FETCH_COIN_SUCCESS:
      return fetchCoinSuccess(state, action)
    case coinAction.ADD_COIN:
      return addCoin(state, action)
    case coinAction.ADD_COIN_SUCCESS:
      return addCoinSuccess(state, action)
    case coinAction.REMOVE_COIN:
      return removeCoin(state, action)
    case coinAction.REMOVE_COIN_SUCCESS:
      return removeCoinSuccess(state, action)
    case coinAction.COIN_ERROR:
      return coinError(state, action)
    default:
      return state
  }
}
