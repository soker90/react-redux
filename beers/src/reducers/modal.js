import modalAction from '../constants/actions/modal'

export function initialState () {
  return {
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}
  }
}

function openModal(state, {modalData}) {
  return {
    ...state,
    open: true,
    title: modalData.title,
    body: modalData.body,
    callback: modalData.callback ? modalData.callback : null,
    extraParams: modalData.extraParams ? modalData.extraParams : {}
  }
}

function closeModal(state) {
  return {
    ...state,
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}

  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case modalAction.OPEN_MODAL:
      return openModal(state, action)
    case modalAction.CLOSE_MODAL:
      return closeModal(state, action)
    default:
      return state
  }
}
