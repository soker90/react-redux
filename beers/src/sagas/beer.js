import { call, put, takeEvery } from 'redux-saga/effects'
import addBeerService from '../services/addBeer'
import fetchBeerService from '../services/fetchBeer'
import removeBeerService from '../services/removeBeer'
import beerActions from '../actions/beer'
import beerActionTypes from '../constants/actions/beer'

export function * fetchBeer () {
  try {
    const response = yield call(fetchBeerService)
    yield put(beerActions.fetchBeerSuccess(response))
  } catch (error) {
    yield put(beerActions.beerError(error))
  }
}

export function * watchFetchBeer () {
  yield takeEvery(beerActionTypes.FETCH_BEER, fetchBeer)
}

export function * addBeer (beer) {
  try {
    const response = yield call(addBeerService, beer)
    yield put(beerActions.addBeerSuccess(response))
  } catch (error) {
    yield put(beerActions.beerError(error))
  }
}

export function * watchAddBeer () {
  yield takeEvery(beerActionTypes.ADD_BEER, addBeer)
}

export function * removeBeer (beerId) {
  try {
    yield call(removeBeerService, beerId)
    yield put(beerActions.removeBeerSuccess(beerId))
  } catch (error) {
    yield put(beerActions.beerError(error))
  }
}

export function * watchRemoveBeer () {
  yield takeEvery(beerActionTypes.REMOVE_BEER, removeBeer)
}
