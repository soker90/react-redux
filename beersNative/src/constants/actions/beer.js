export default {
  FETCH_BEER: 'FETCH_BEER',
  FETCH_BEER_SUCCESS: 'FETCH_BEER_SUCCESS',
  ADD_BEER: 'ADD_BEER',
  ADD_BEER_SUCCESS: 'ADD_BEER_SUCCESS',
  REMOVE_BEER: 'REMOVE_BEER',
  REMOVE_BEER_SUCCESS: 'REMOVE_BEER_SUCCESS',
  BEER_ERROR: 'BEER_ERROR'
}
