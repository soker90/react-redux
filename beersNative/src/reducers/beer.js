import beerAction from '../constants/actions/beer'

export function initialState () {
  return {
    beerList: [],
    beer: {},
    loading: false,
    error: null
  }
}

function fetchBeer(state) {
  return {
    ...state,
    beerList: [],
    loading: true,
    error: null
  }
}

function fetchBeerSuccess(state, {beers}) {
  return {
    ...state,
    beerList: beers.beers,
    loading: false,
    error: null
  }
}

function addBeer(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function addBeerSuccess(state, {beer}) {
  return {
    ...state,
    beerList: [beer, ...state.beerList],
    loading: false,
    error: null
  }
}

function removeBeer(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function removeBeerSuccess(state, {beerId}) {
  return {
    ...state,
    beerList: state.beerList.filter((beer) => beer.id !== beerId.beerId),
    loading: false,
    error: null
  }
}

function beerError(state, {error}) {
  return {
    ...state,
    loading: false,
    error: error
  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case beerAction.FETCH_BEER:
      return fetchBeer(state, action)
    case beerAction.FETCH_BEER_SUCCESS:
      return fetchBeerSuccess(state, action)
    case beerAction.ADD_BEER:
      return addBeer(state, action)
    case beerAction.ADD_BEER_SUCCESS:
      return addBeerSuccess(state, action)
    case beerAction.REMOVE_BEER:
      return removeBeer(state, action)
    case beerAction.REMOVE_BEER_SUCCESS:
      return removeBeerSuccess(state, action)
    case beerAction.BEER_ERROR:
      return beerError(state, action)
    default:
      return state
  }
}
