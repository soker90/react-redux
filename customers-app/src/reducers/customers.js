import { handleActions } from 'redux-actions'
import { FETCH_CUSTOMERS, INSERT_COSTUMER } from '../constants'

export const customers = handleActions({
	[FETCH_CUSTOMERS]: (state, action) => [...action.payload],
	[INSERT_COSTUMER]: (state, action) => [...state, action.payload],
}, [])

