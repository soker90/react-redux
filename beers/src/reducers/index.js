import { combineReducers } from 'redux'
import beers from './beer'
import modal from './modal'
import coins from './coin'
import { reducer as formReducer } from 'redux-form'

const rootReducer = combineReducers({
  beers,
  modal,
  coins,
  form: formReducer
})

export default rootReducer
