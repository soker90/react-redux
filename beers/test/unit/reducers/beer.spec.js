// beer.spec.js
import beerAction from '../../../src/actions/beer'
import beerReducer from '../../../src/reducers/beer'
import { initialState } from '../../../src/reducers/beer'
import mockData from '../mockData'

describe('Beer reducer', () => {

  test('should handle initial state', () => {
    expect(initialState()).toMatchSnapshot()
  })

  test('fetchBeer method should handle show loading', () => {
    const result = beerReducer({}, beerAction.fetchBeer())

    expect(result).toMatchSnapshot()
  })

  test('fetchBeerSuccess method should handle hide loading and load beers', () => {
    const result = beerReducer(initialState(), beerAction.fetchBeerSuccess(mockData.beers))

    expect(result).toMatchSnapshot()
  })

  test('addBeer method should handle show loading', () => {
    const result = beerReducer(initialState(), beerAction.addBeer())

    expect(result).toMatchSnapshot()
  })

  test('addBeerSuccess method should handle hide loading and add beer', () => {
    const result = beerReducer(initialState(), beerAction.addBeerSuccess(mockData.beer))

    expect(result).toMatchSnapshot()
  })

  test('removeBeer method should handle show loading', () => {
    const result = beerReducer(initialState(), beerAction.removeBeer())

    expect(result).toMatchSnapshot()
  })

  test('removeBeerSuccess method should handle hide loading and remove beer', () => {
    const state = {
      beerList: mockData.beers,
      error: null,
      loading: false
    }
    const result = beerReducer(state, beerAction.removeBeerSuccess(mockData.beerId))

    expect(result).toMatchSnapshot()
  })
})
