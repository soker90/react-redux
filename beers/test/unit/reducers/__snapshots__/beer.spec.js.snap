exports[`Beer reducer addBeer method should handle show loading 1`] = `
Object {
  "beer": Object {},
  "beerList": Array [],
  "error": null,
  "loading": true,
}
`;

exports[`Beer reducer addBeerSuccess method should handle hide loading and add beer 1`] = `
Object {
  "beer": Object {},
  "beerList": Array [
    Object {
      "graduation": "5",
      "id": "beer_01",
      "name": "Amstel",
    },
  ],
  "error": null,
  "loading": false,
}
`;

exports[`Beer reducer fetchBeer method should handle show loading 1`] = `
Object {
  "beerList": Array [],
  "error": null,
  "loading": true,
}
`;

exports[`Beer reducer fetchBeerSuccess method should handle hide loading and load beers 1`] = `
Object {
  "beer": Object {},
  "beerList": undefined,
  "error": null,
  "loading": false,
}
`;

exports[`Beer reducer removeBeer method should handle show loading 1`] = `
Object {
  "beer": Object {},
  "beerList": Array [],
  "error": null,
  "loading": true,
}
`;

exports[`Beer reducer removeBeerSuccess method should handle hide loading and remove beer 1`] = `
Object {
  "beerList": Array [
    Object {
      "graduation": "5",
      "id": "beer_01",
      "name": "Amstel",
    },
    Object {
      "graduation": "3",
      "id": "beer_02",
      "name": "Mahoo",
    },
    Object {
      "graduation": "4,5",
      "id": "beer_03",
      "name": "Corona",
    },
  ],
  "error": null,
  "loading": false,
}
`;

exports[`Beer reducer should handle initial state 1`] = `
Object {
  "beer": Object {},
  "beerList": Array [],
  "error": null,
  "loading": false,
}
`;
